apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c

```diff
@@ -0,0 +1,848 @@
+*/
+static void utc_audio_pcm_open_tc_n(void)
+{
+	struct pcm *pcm;
+	pcm = pcm_open(999, 999, PCM_IN, NULL);
+	TC_ASSERT_LEQ("pcm_open", pcm_get_file_descriptor(pcm), 0);
+	pcm_close(pcm);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_is_ready_p
+* @brief            open and check pcm is ready or not
+* @scenario         Open pcm and check its state
+* @apicovered       pcm_is_ready
+* @precondition     pcm_open should be called
+* @postcondition    NA
+*/
+static void utc_audio_pcm_is_ready_tc_p(void)
+{
+	TC_ASSERT("pcm_is_ready", pcm_is_ready(g_pcm));
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_is_ready_n
+* @brief            open and check pcm is ready or not
+* @scenario         Open pcm and check its state with NULL pcm
+* @apicovered       pcm_is_ready
+* @precondition     pcm_open should be called
+* @postcondition    NA
+*/
+static void utc_audio_pcm_is_ready_tc_n(void)
+{
+	TC_ASSERT_LEQ("pcm_is_ready", pcm_is_ready(NULL), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_prepare_p
+* @brief            open and prepare pcm
+* @scenario         Open pcm and prepare
+* @apicovered       pcm_prepare
+* @precondition     pcm_open should be called
+* @postcondition    NA
+*/
+static void utc_audio_pcm_prepare_tc_p(void)
+{
+	TC_ASSERT_EQ("pcm_prepare", pcm_prepare(g_pcm), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_prepare_n
+* @brief            open and prepare pcm
+* @scenario         Open pcm and prepare with NULL pcm
+* @apicovered       pcm_prepare
+* @precondition     pcm_open should be called
+* @postcondition    NA
+*/
+static void utc_audio_pcm_prepare_tc_n(void)
+{
+	TC_ASSERT_LT("pcm_prepare", pcm_prepare(NULL), 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_close_p
+* @brief            close opened pcm
+* @scenario         close opened pcm
+* @apicovered       pcm_close
+* @precondition     pcm should be opened before.
+* @postcondition    NA
+*/
+static void utc_audio_pcm_close_tc_p(void)
+{
+	int ret;
+	ret = pcm_close(g_pcm);
+	TC_ASSERT_EQ("pcm_close", ret, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_close_n
+* @brief            close opened pcm
+* @scenario         close opened pcm with null pcm
+* @apicovered       pcm_close
+* @precondition     NA
+* @postcondition    NA
+*/
+static void utc_audio_pcm_close_tc_n(void)
+{
+	int ret;
+	ret = pcm_close(NULL);
+	TC_ASSERT_NEQ("pcm_close", ret, 0);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_open_by_name_p
+* @brief            open with name and initlaize pcm structure
+* @scenario         Open pcm and check its validation
+* @apicovered       pcm_open_by_name
+* @precondition     NA
+* @postcondition    NA
+*/
+static void utc_audio_pcm_open_by_name_tc_p(void)
+{
+	g_pcm = pcm_open_by_name("hw:0,0", PCM_IN, NULL);
+	TC_ASSERT_GT("pcm_open_by_name", pcm_get_file_descriptor(g_pcm), 0)
+	TC_SUCCESS_RESULT();
+	pcm_close(g_pcm);
```

Comment:
At TC SUCCESS RESULT, this function will be returned. Move this line

Reply:
done